gossipd/routing.c

```diff
@@ -573,8 +573,8 @@ bool handle_channel_announcement(
 	/* Is this a new connection? It is if we don't know the
 	 * channel yet, or do not have a matching announcement in the
 	 * case of side-loaded channels*/
-	c0 = get_connection_by_scid(rstate, &short_channel_id, 0);
-	c1 = get_connection_by_scid(rstate, &short_channel_id, 1);
+	c0 = get_connection(rstate, &node_id_2, &node_id_1);
+	c1 = get_connection(rstate, &node_id_1, &node_id_2);
```

Comment:
I'm confused.  There can be multiple channels between two nodes.  I'll apply this since you indicated it's a workaround, but it seems we'll not have the correct topology for this case...